pyspark/dl/optim/optimizer.py

```diff
@@ -188,6 +188,14 @@ def setvalidation(self, batch_size, val_rdd, trigger, val_method=["Top1Accuracy"
         callBigDlFunc(self.bigdl_type, "setValidation", self.value, batch_size,
                       trigger, val_rdd, val_method)
 
+    def setmodel(self, model):
```

Comment:
Again, I think optimizer should not cache the data. It's not a job of the optimizer.

I'm agreed with your opinion. A prepare_input is misleading. I think the problem mentioned in the issue only a problem in some interactive programming(e.g. notebook), user may feel it takes long time when execute create optimizer. It's not a problem in a batch execution.

A prepare_input allow notebook user to seperate the data loading time and training time. If you don't have it, user may think the training take so long time, in which most is spent on loading data. I think maybe we provide this method in optimizer, even user didn't call that method, optimizer will do it at the start of training. And don't cache the data directly in optimizer. A dataset should also provide a method prepareData to hide the details.